functests/xt_u32/xt_u32.go

```diff
@@ -0,0 +1,223 @@
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Check if the pod is running")
+			Eventually(func() k8sv1.PodPhase {
+				res, err := client.Client.Pods(TestNamespace).Get(context.Background(), xt_u32Pod.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return res.Status.Phase
+			}, 3*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodRunning))
+
+			By("Check that xt_u32 module is loaded")
+			cmd := []string{"lsmod"}
+			out, err := pods.ExecCommand(client.Client, *xt_u32Pod, cmd)
+			Expect(err).ToNot(HaveOccurred())
+			res := strings.Contains(out.String(), "xt_u32")
+			if !res {
+				err = errors.New("xt_u32 module is not loaded")
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			By("Create an iptables rule within the pod")
+			cmd = []string{"iptables", "-t", "filter", "-A", "INPUT", "-i", "eth0", "-m",
+				"u32", "--u32", "6 & 0xFF = 1 && 4 & 0x3FFF = 0 && 0 >> 22 & 0x3C @ 0 >> 24 = 8",
+				"-j", "DROP"}
+			_, err = pods.ExecCommand(client.Client, *xt_u32Pod, cmd)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+})
+
+func findXT_U32NodeSelector() (string, error) {
+	mcList, err := client.Client.MachineConfigs().List(context.Background(), metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, mc := range mcList.Items {
+		enables, err := enablesXT_U32(mc)
+		if err != nil {
+			return "", err
+		}
+		if enables {
+			mcLabel, found := mc.ObjectMeta.Labels["machineconfiguration.openshift.io/role"]
+			if !found {
+				continue
+			}
+
+			xt_u32NodeSelector, err := findXT_U32NodeSelectorByMCLabel(mcLabel)
+			if err != nil {
+				continue
+			}
+
+			return xt_u32NodeSelector, nil
+		}
+	}
+
+	return "", errors.New("Cannot find a machine configuration that enables XT_U32")
+}
+
+func enablesXT_U32(mc mcfgv1.MachineConfig) (bool, error) {
+	loadPathFound := false
+
+	ignitionConfig := igntypes.Config{}
+	if mc.Spec.Config.Raw == nil {
+		return false, nil
+	}
+
+	err := json.Unmarshal(mc.Spec.Config.Raw, &ignitionConfig)
+	if err != nil {
+		return false, fmt.Errorf("Failed to unmarshal ignition config %v", err)
+	}
+
+	for _, file := range ignitionConfig.Storage.Files {
+		if !loadPathFound && file.Path == xt_u32LoadPath {
+			loadPathFound = true
+		}
+	}
+	return loadPathFound, nil
+}
+
+func findXT_U32NodeSelectorByMCLabel(mcLabel string) (string, error) {
+	mcpList, err := client.Client.MachineConfigPools().List(context.Background(), metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, mcp := range mcpList.Items {
+		for _, lsr := range mcp.Spec.MachineConfigSelector.MatchExpressions {
+			for _, value := range lsr.Values {
+				if value == mcLabel {
+					for key, label := range mcp.Spec.NodeSelector.MatchLabels {
+						newXT_U32NodeSelector := key + "=" + label
+						return newXT_U32NodeSelector, nil
+					}
+				}
+			}
+		}
+	}
+
+	return "", errors.New("Cannot find XT_U32NodeSelector")
+}
+
+func xt_u32TestPod(name, node, namespace string, args []string) *k8sv1.Pod {
```

Comment:
since that a privileged pod, I clearly state that in the function name,
so anyone using it, will know that.
we probably need a naming convention for privileged operations too?

Reply:
i can rename the function to xt_u32PrivilegedTestPod
what privileged operations are you referring to?